src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -101,11 +100,66 @@ internal WebApplicationBuilder(Assembly? callingAssembly, string[]? args = null)
         /// <returns>A configured <see cref="WebApplication"/>.</returns>
         public WebApplication Build()
         {
+            // Copy the configuration sources into the final IConfigurationBuilder
+            _hostBuilder.ConfigureHostConfiguration(builder =>
+            {
+                foreach (var source in ((IConfigurationBuilder)Configuration).Sources)
+                {
+                    builder.Sources.Add(source);
```

Comment:
I'm sad we lose the ability to observe changes to sources post-build.

Reply:
We probably need to make sure that works.

Reply:
I should have added a test case for it, but I was worried you'd balk at requiring that behavior. I didn't think you'd merge it so quickly 😅

Reply:
Actually it works fine.